src/engine/renderer/tr_bsp.cpp

```diff
@@ -955,7 +955,24 @@ static void ParseFace( dsurface_t *ds, drawVert_t *verts, bspSurface_t *surf, in
 	for( i = 0; i < numVerts; i++ ) {
 		if( components[ i ].minVertex == i ) {
 			for( j = 0; j < 2; j++ ) {
-				components[ i ].stBounds[ 0 ][ j ] = rintf( 0.5f * (components[ i ].stBounds[ 1 ][ j ] + components[ i ].stBounds[ 0 ][ j ]) );
+				/* Some words about the “minus 0.5” trick:
+				 *
+				 * The vertexpack engine tries to optimize texture coords, because fp16 numbers have 1/2048 resolution only between -1.0 and 1.0,
+				 * and so we could have several texel rounding errors if the coords are too large. The optimization finds connected surfaces and
+				 * then shifts the texture coords so that their average value is near 0. If the range is 0-1, there are two equally good solutions,
+				 * either keep the range 0-1 or shift to -1-0, and in this case tiny rounding errors decide which one is used.
+				 * For non-animated textures this is no issue, but the rotating textures always rotate around 0.5/0.5, and if the texture coords
+				 * are shifted, it shifts the rotation center too. The easy fix is to move the average as close as possible to 0.5 instead of 0.0,
+				 * then all standard 0-1 textures will not be shifted.
+				 * -- @gimhael https://github.com/DaemonEngine/Daemon/issues/35#issuecomment-507406783
+				 *
+				 * Instead of round(x - 0.5) we can do floor(x). (using std::floorf)
```

Comment:
I don't really think my comment was useful enough to be worth putting in the code (this kind of detail about what the compiler emits can change over time), but if you decide to keep it, there's an issue with a missing footnote.

Reply:
Yeah, I was doing a footnote then I thought the link was absolutely useless. I just removed the footnote index.
I think your comment is useful in the way it explains why this solution was chosen instead of others at the time it was done.

Such operations with so much implicit and collateral behaviors looks like hacks when viewed from an unadvised point of view, that's why I think it's precious to left comments that explains how such choice was motivated at this time with serious arguments.

This reminds me what happened with: https://github.com/DaemonEngine/Daemon/issues/183#issuecomment-473691252
At first look it looked weird and asking for a fix, but in fact that was deeply thought.

I find useful to tell future people why the more cryptic from `rintf(x - 0.5f)` was chosen over simpler `floorf(x)`: readability was traded against performance, and such thing have to be motivated. :-)